sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java

```diff
@@ -433,7 +435,22 @@ public synchronized String toString() {
    */
   private Object getValueFromJson(String propertyName, Method method) {
     try {
-      JavaType type = MAPPER.getTypeFactory().constructType(method.getGenericReturnType());
+      JavaType type;
+      Type returnType = method.getReturnType();
+      if (returnType.equals(ValueProvider.class)) {
+        if (returnType instanceof ParameterizedType) {
```

Comment:
returnType is either a Class or a ParameterizedType, I believe. The only case in which it would be equal to ValueProvider.class is if it was a class.